crates/rolldown/src/types/generator.rs

```diff
@@ -17,6 +19,59 @@ pub struct GenerateContext<'a> {
   pub module_id_to_codegen_ret: Vec<Option<ModuleRenderOutput>>,
 }
 
+impl<'a> GenerateContext<'a> {
+  /// A `SymbolRef` might be identifier or a property access. This function will return correct string pattern for the symbol.
+  pub fn finalized_string_pattern_for_symbol_ref(
+    &self,
+    symbol_ref: SymbolRef,
+    cur_chunk_idx: ChunkIdx,
+    canonical_names: &FxHashMap<SymbolRef, Rstr>,
+  ) -> String {
+    let symbol_db = &self.link_output.symbol_db;
+    // let belong_to_chunk_idx =
+    // if !symbol_ref.is_declared_in_root_scope(self.ctx.symbol_db) {
+    //   // No fancy things on none root scope symbols
+    //   return self.snippet.id_ref_expr(self.canonical_name_for(symbol_ref), SPAN);
+    // }
+    let canonical_ref = symbol_db.canonical_ref_for(symbol_ref);
+    let canonical_symbol = symbol_db.get(canonical_ref);
+    let namespace_alias = &canonical_symbol.namespace_alias;
+    if let Some(_ns_alias) = namespace_alias {
+      // canonical_ref = ns_alias.namespace_ref;
+      // canonical_symbol = symbol_db.get(canonical_ref);
+      unreachable!("Not sure if we need to handle this case");
```

Comment:
Can you use a more formal message? It might be confused for user saw such a error meesage.